HpToolsLauncher/TestRunners/ApiTestRunner.cs

```diff
@@ -72,7 +72,36 @@ public TestRunResults RunTest(TestInfo testinf, ref string errorReason, RunCance
             ConsoleWriter.ActiveTestRun = runDesc;
             ConsoleWriter.WriteLine(DateTime.Now.ToString(Launcher.DateFormat) + " Running: " + testinf.TestPath);
 
-            runDesc.ReportLocation = testinf.TestPath;
+            // check if the report path has been defined
+            if (!String.IsNullOrEmpty(testinf.ReportPath))
+            {
+                runDesc.ReportLocation = testinf.ReportPath;
+
+                if (!Directory.Exists(runDesc.ReportLocation))
+                {
+                    try
+                    {
+                        Directory.CreateDirectory(runDesc.ReportLocation);
+                    }
+                    catch (Exception)
+                    {
```

Comment:
Extract the catch content into a function to provide an additional explanation about its purpose